### PR TITLE
WIP: Tests for Metacom Server

### PR DESCRIPTION
**Pull request by georgolden**
- [ ] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [ ] code is properly formatted (`npm run fmt`)
- [ ] description of changes is added in CHANGELOG.md
- [ ] update .d.ts typings

### CODE REVIEW

**Review by georgolden — COMMENTED** [review]
This behavior exists with balancer too. It just spawns servers in other workers with ports undefined.

File: test/server.js [hunk 1]
```diff
@@ -1,3 +1,51 @@
...
+    queue: {
+      concurrency: 1000,
+      size: 2000,
+      timeout: 3000,
+    },
+    workers: {
+      pool: 2,
+    },
+  };
+  const application = {
```
Comment (tshemsedinov) [inline test/server.js:42]:
```suggestion

  const application = {
```

File: test/worker.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const { parentPort } = require('worker_threads');
+const { Server } = require('../lib/server');
+class Console {
```
Comment (tshemsedinov) [inline test/worker.js:5]:
```suggestion

class Console {
```

File: test/worker.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  queue: {
+    concurrency: 1000,
+    size: 2000,
+    timeout: 3000,
+  },
+  workers: {
+    pool: 2,
+  },
+};
+const application = {
```
Comment (tshemsedinov) [inline test/worker.js:46]:
```suggestion

const application = {
```

File: test/worker.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  console: new Console(),
+};
+
+(async () => {
+  await new Server(config, application);
+  application.console.infoData.forEach((data) => {
+    parentPort.postMessage(data);
+  });
```
Comment (tshemsedinov) [inline test/worker.js:54]:
```suggestion
  for (const data of application.console.infoData) {
    parentPort.postMessage(data);
  }
```

File: test/master.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+const metatests = require('metatests');
+const { Worker } = require('worker_threads');
+
+metatests.test(
+  'Start server without balancer with 3 workers 2 ports',
+  (test) => {
+    const workersNum = 3;
+
+    for (let i = 0; i < workersNum; i++) {
```
Comment (tshemsedinov) [inline test/master.js:10]:
```suggestion
    for (let i = 1; i < workersNum; i++) {
```
- Number of workers should be equal to number of ports
- Thread 0 in intended for system use, so we will skip it

**Review by tshemsedinov — COMMENTED** [review]
- There is not problem if no balancer configured.
- We don't use semantic commits in this repo
- Please use --fixup for changing something to be automatically squashed

File: test/master.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+const metatests = require('metatests');
```
Comment (tshemsedinov) [inline test/master.js:2]:
```suggestion

const metatests = require('metatests');
```

File: test/master.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+const { Worker } = require('worker_threads');
+
+metatests.test(
+  'Start server without balancer with 3 workers 2 ports',
```
Comment (tshemsedinov) [inline test/master.js:6]:
```suggestion
  'Start server without balancer with 2 workers (2 ports)',
```

File: test/server.js [hunk 8]
```diff
@@ -1,3 +1,52 @@
 'use strict';
+const metatests = require('metatests');
```
Comment (tshemsedinov) [inline test/server.js:2]:
```suggestion

const metatests = require('metatests');
```

File: test/server.js [hunk 9]
```diff
@@ -1,3 +1,52 @@
...
+  const application = {
+    console: new Console(),
+  };
+
+  await new Server(config, application);
+
+  test.strictSame(application.console.infoData, 'Listen port 8000 in worker 0');
+
+  test.end();
+});
```
Comment (tshemsedinov) [inline test/server.js:52]:
```suggestion
console.log('No server tests implemented');
```

File: test/server.js [hunk 10]
```diff
@@ -1,3 +1,6 @@
 'use strict';
 
+const metatests = require('metatests');
+const { Server } = require('../lib/server');
+
```
Comment (tshemsedinov) [inline test/server.js:5]:
```suggestion
```

**Pull request closed without merging**